blade-formatter: init at 1.41.1

lelgenio opened the pull request:
## Description of changes

A formatter for Blade templates from Laravel PHP framework.

https://github.com/shufo/blade-formatter

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->


Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

Commit:
blade-formatter: init at 1.41.1

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2737

heisfer commented on pkgs/by-name/bl/blade-formatter/package.nix:
```diff
@@ -0,0 +1,61 @@
+    };
+  };
+
+  meta = with lib; {
```
Remove `with lib` it's antipattern. Use `with lib.maintainers` etc below.

heisfer commented on pkgs/by-name/bl/blade-formatter/package.nix:
```diff
@@ -0,0 +1,61 @@
+
+  src = fetchFromGitHub {
+    owner = "shufo";
+    repo = pname;
```
Replace it with "blade-formatter".

lelgenio commented:
Things done:

- applied suggestions
- changed from `mkYarnPackage` to `buildNpmPackage`
- added an update script
- updated from `1.38.2` to `1.41.1`

heisfer commented:
Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

heisfer commented:
Just a few additional notes.
1. `meta.mainProgram = "blade-formatter";` (so that it can be used with `lib.getExe pkgs.blade-formatter` instead of `lib.getExe' pkgs blade-formatter "blade-formatter"`
2. format with [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=nixfmt-rfc) (seems to be the new trend)


Tested basic blade file and worked as expected.

```
<x-layout> <x-slot:heading>Job</x-slot:heading> <h2 class="font-bold text-lg">{{ $job->title }}</h2> <p>Salary: {{$job->salary}}</p> @can('edit', $job) <p class="mt-6"> <x-button href="/jobs/{{ $job->id }}/edit">Edit Job</x-button> </p> @endcan </x-layout>
```
```
<x-layout> <x-slot:heading>Job</x-slot:heading>
    <h2 class="font-bold text-lg">{{ $job->title }}</h2>
    <p>Salary: {{ $job->salary }}</p> @can('edit', $job)
        <p class="mt-6"> <x-button href="/jobs/{{ $job->id }}/edit">Edit Job</x-button> </p>
    @endcan
</x-layout>
```

heisfer reviewed (approved):
LGTM

lelgenio commented:
Oops, fumbled `meta.platforms` :confounded:

lelgenio commented:
Result of `nixpkgs-review pr 260000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>blade-formatter</li>
  </ul>
</details>

pbsds commented on pkgs/by-name/bl/blade-formatter/update.sh:
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p ripgrep common-updater-scripts prefetch-npm-deps jq
```
```suggestion
#!nix-shell -i bash -p ripgrep common-updater-scripts prefetch-npm-deps jq sd
```

lelgenio replied:
Thanks :smiley: :+1:

pbsds commented:
Congrats on the round PR number btw :tada:  :tada:  :tada:
